### DISCUSSION TITLE
YouTube Video Summarization with YouLama

### DISCUSSION THREAD

**Discussion by tcsenpai** [dailydotdev/daily#1660] [Feature Requests]
Hi!
I am using since months my https://github.com/tcsenpai/youlama to summarize YouTube videos posted on Daily Dev. I find it extremely useful for use cases like:
- Clickbait titles
- Long videos that I can't watch at that moment
- Having a decent preview of what I am going to watch

It is a fairly simple piece of software that works nicely even with small (8b) models and might work with even smaller ones if optimized enough.

I was wondering if it would be possible to use it as a feature in Daily Dev. If it is too resource intensive, could be a Plus feature. Of course anyone could just download and use YouLama locally, provided the GPU power to use it or the patience to use it with CPU, but I think this could be a nice feature.

If any of the maintainers want to try it on my instance, I can share it with them. I don't post my public instance because it would cause my home PC to explode probably. But yeah, if this can help I would be happy.

**Comment by idoshamun** [comment]
Hi @tcsenpai,

Thanks a lot for the suggestion. Have you noticed we have a summary already?

<img width="639" alt="image" src="https://github.com/user-attachments/assets/c82dd29e-5cca-42fd-91c7-f6a6ce25a379" />

**Comment by tcsenpai** [comment]
Yep absolutely. That's what it inspired me in the first place to suggest this.
YouLama is more detailed and tailored to long videos (e.g. summarizing 30-min videos in a couple of paragraphs), hence the (I guess) more resource intensive inner working. Your TLDR is indeed perfect for the preview, while YouLama is often way more detailed (and verbose, so probably not fit for the preview).

I am posting the YouLama output for the video above for comparison in the next reply.

**Reply by idoshamun** [reply]
We use GPT models to generate the summary. Regarding the suggestion for a longer summary, we are going to introduce a new feature as part of the plus subscription that allows you to run custom prompts on the post (text/video/etc). I think this will be a perfect use case for that

**Reply by tcsenpai** [reply]
Yep, it looks like it is a perfect match and is integrating the very use case I am suggesting. Thanks for the swift reply and really thank you a lot for the efforts in building what's my home page basically : )

**Comment by tcsenpai** [comment]
Video Information:

Title: So I Tried To Learn Shaders...

Channel: ThePrimeTime
Summary:
### Shader Summarization

Welcome to the fascinating world of shaders! This document summarizes a discussion on various aspects of shaders, including easing functions, data structures, and practical applications. Let's dive right into it.
### Easing Functions

The discussion begins with an exploration of easing functions, which are mathematical functions that help control the acceleration and deceleration of animated objects. Participants noted their familiarity with these functions, recognizing parabolas and S-curves among them. The importance of understanding these functions is emphasized, particularly in specialized fields such as game development.
### Practical Applications

Participants experimented with drawing triangles and other shapes using shaders. They recognized the triangular nature of the drawings and practiced replicating various equations to improve their skills. The goal was to master these techniques through repetition, much like training in karate.
### Data Structures

The conversation then delved into more complex data structures, including SDF (Signed Distance Field). Although some participants were unfamiliar with this concept initially, they acknowledged the need for extensive practice to fully grasp it. The discussion highlighted the mathematical foundations of sine and cosine functions and their roles in creating shapes like circles.
### Challenges and Next Steps

Despite the enthusiasm, participants faced challenges in understanding more advanced topics. They expressed frustration at not being experts immediately, highlighting the importance of persistent practice. The discussion ended on a positive note, with plans to continue exploring shaders further, particularly drawing complex shapes like sparkles.
### Conclusion

Shader programming is an intricate field that demands patience and persistence. Through hands-on practice and thorough understanding of easing functions and data structures, participants can significantly improve their skills. Whether you're new to shaders or seeking to refine your expertise, the journey is both challenging and rewarding. Happy coding!
